libnavigation-core/src/main/java/com/mapbox/navigation/core/preview/RoutesPreview.kt

```diff
@@ -0,0 +1,42 @@
+package com.mapbox.navigation.core.preview
+
+import androidx.annotation.StringDef
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.routealternatives.AlternativeRouteMetadata
+
+@ExperimentalPreviewMapboxNavigationAPI
+data class RoutesPreview(
+    val primaryRoute: NavigationRoute,
+    val alternatives: List<AlternativeRoutePreview>,
+    val routesList: List<NavigationRoute>,
+    val originalRoutesList: List<NavigationRoute>
+)
+
+@ExperimentalPreviewMapboxNavigationAPI
+data class AlternativeRoutePreview(
+    val alternative: NavigationRoute,
+    val metadata: AlternativeRouteMetadata
+)
+
+@ExperimentalPreviewMapboxNavigationAPI
+data class RoutesPreviewUpdate(
+    @RoutePreviewUpdateReason val reason: String,
+    val routePreview: RoutesPreview?
+)
+
+object RoutesPreviewUpdateReasons {
+    const val PREVIEW_NEW = "PREVIEW_SET"
+    const val PREVIEW_CLEAN_UP = "PREVIEW_CLEAN_UP"
+}
+
+@StringDef(
+    RoutesPreviewUpdateReasons.PREVIEW_NEW,
+    RoutesPreviewUpdateReasons.PREVIEW_CLEAN_UP
+)
+annotation class RoutePreviewUpdateReason
```

Comment:
Move this to a separate file and add comments to describe what exactly the reason means. use [RoutesExtra](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesExtra.kt) and [RouteRefreshExtra](https://github.com/mapbox/mapbox-navigation-android/blob/main/libnavigation-core/src/main/java/com/mapbox/navigation/core/routerefresh/RouteRefreshExtra.kt) as an example

Reply:
Is there a reason to do this different that RoutesExtra and RoutesRefreshExtra? Nick picking here but it seems we might as well make them consistent right?

Reply:
renamed to `RoutesPreviewExtra`